lib/modules/s3.rb

```diff
@@ -30,23 +30,17 @@ def self.link_to file_name, opts={for_import: false}
   end
 
   def upload object_name, source, opts
-    # Check to see what environment Rails is currently running in
-    bucket = Rails.application.secrets.aws_downloads_bucket
-
-    if Rails.env.staging? || Rails.env.production?
-      bucket = Rails.application.secrets.aws_files_bucket
-    end
+    # Default to downloads bucket unless specified (e.g. for when uploading CMS files)
+    bucket = opts[:bucket] || Rails.application.secrets.aws_downloads_bucket
```

Comment:
It looks you've partly reverted my change. Did it not work with the secrets?

Reply:
So this method have been used to upload downloads files on S3 so far, so using the aws_downloads_bucket.
Now it would also need to use another bucket for uploading the CMS files, so yeah it was necessary to make this change in the end.

With your latest code though, it was using the aws_downloads_bucket on staging or production regardless if the upload intent was for CMS files or for downloads; while using the files bucket by default if it was development instead